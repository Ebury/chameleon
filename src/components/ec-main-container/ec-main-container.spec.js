import { mount } from '@vue/test-utils';

import EcMainContainer from './ec-main-container.vue';

describe('EcMainContainer', () => {
  it('should render empty if no props were given', () => {
    const wrapper = mount(EcMainContainer);

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render both titles when they are given in props', () => {
    const props = {
      title: 'Trade Finance',
      titleIntro: 'Here you will be able to keep track of all your requests to Ebury and your credit line.',
    };
    const wrapper = mount(EcMainContainer, {
      props,
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should not display the titleIntro when title is not given', () => {
    const props = {
      titleIntro: 'Here you will be able to keep track of all your requests to Ebury and your credit line.',
    };
    const wrapper = mount(EcMainContainer, {
      props,
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render without title intro when the titleIntro prop is not given', () => {
    const props = {
      title: 'Trade Finance',
    };
    const wrapper = mount(EcMainContainer, {
      props,
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should use given slot', () => {
    const props = {
      title: 'Trade Finance',
      titleIntro: 'Here you will be able to keep track of all your requests to Ebury and your credit line.',
    };
    const wrapper = mount(EcMainContainer, {
      props,
      slots: {
        default: '<p>Random text</p>',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should use the given named cta slot', () => {
    const props = {
      title: 'Trade Finance',
      titleIntro: 'Here you will be able to keep track of all your requests to Ebury and your credit line.',
    };
    const wrapper = mount(EcMainContainer, {
      props,
      slots: {
        cta: '<button class="ec-btn ec-btn--rounded ec-btn--primary ec-btn--md ec-btn--full-width">Test cta</button>',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should use the given named breadcrumbs slot', () => {
    const props = {
      title: 'Trade Finance',
      titleIntro: 'Here you will be able to keep track of all your requests to Ebury and your credit line.',
    };
    const wrapper = mount(EcMainContainer, {
      props,
      slots: {
        breadcrumbs: '<a href="#">Breadcrumbs</a>',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
