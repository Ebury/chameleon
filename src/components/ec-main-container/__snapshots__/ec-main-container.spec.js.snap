// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcMainContainer should not display the titleIntro when title is not given 1`] = `
<div
  class="ec-main-container"
>
  <!---->
   
</div>
`;

exports[`EcMainContainer should render both titles when they are given in props 1`] = `
<div
  class="ec-main-container"
>
  <div
    class="tw-grid"
  >
    <div
      class="tw-col-full"
    >
      <h1
        class="ec-main-container__title"
      >
        Trade Finance
      </h1>
       
      <p
        class="ec-main-container__title-intro"
      >
        Here you will be able to keep track of all your requests to Ebury and of your credit line.
      </p>
    </div>
     
    <!---->
  </div>
   
</div>
`;

exports[`EcMainContainer should render empty if no props were given 1`] = `
<div
  class="ec-main-container"
>
  <!---->
   
</div>
`;

exports[`EcMainContainer should use given slot 1`] = `
<div
  class="ec-main-container"
>
  <div
    class="tw-grid"
  >
    <div
      class="tw-col-full"
    >
      <h1
        class="ec-main-container__title"
      >
        Trade Finance
      </h1>
       
      <p
        class="ec-main-container__title-intro"
      >
        Here you will be able to keep track of all your requests to Ebury and of your credit line.
      </p>
    </div>
     
    <!---->
  </div>
   
  <p>
    Random text
  </p>
</div>
`;

exports[`EcMainContainer should use the given named cta slot 1`] = `
<div
  class="ec-main-container"
>
  <div
    class="tw-grid"
  >
    <div
      class="tw-col-full sm:tw-col-9"
    >
      <h1
        class="ec-main-container__title"
      >
        Trade Finance
      </h1>
       
      <p
        class="ec-main-container__title-intro"
      >
        Here you will be able to keep track of all your requests to Ebury and of your credit line.
      </p>
    </div>
     
    <div
      class="tw-col-full sm:tw-col-3"
    >
      <button
        class="ec-btn ec-btn--rounded ec-btn--primary ec-btn--md ec-btn--full-width"
      >
        Test cta
      </button>
    </div>
  </div>
   
</div>
`;
